tests/integration/test_kubernetes_worker_integration.py

```diff
@@ -113,19 +114,28 @@ async def juju_run(unit, cmd):
 
 
 async def test_kube_api_endpoint(ops_test):
-    """ Validate that adding the kube-api-endpoint relation works """
+    """Validate that adding the kube-api-endpoint relation works"""
     await ops_test.model.add_relation(
-        "kubernetes-master:kube-api-endpoint", "kubernetes-worker:kube-api-endpoint"
+        "kubernetes-control-plane:kube-api-endpoint",
+        "kubernetes-worker:kube-api-endpoint",
     )
 
     # It can take some time for the relation hook to trigger, which can lead to
     # wait_for_idle giving a false positive.
-    k8s_master = ops_test.model.applications["kubernetes-master"].units[0]
+    k8s_cp = ops_test.model.applications["kubernetes-control-plane"].units[0]
     waiting_msg = "Waiting for kube-api-endpoint relation"
-    await ops_test.model.block_until(
-        lambda: k8s_master.workload_status_message == waiting_msg,
-        timeout=5 * 60,
-    )
+    try:
+        await ops_test.model.block_until(
+            lambda: k8s_cp.workload_status_message == waiting_msg,
+            timeout=5 * 60,
+        )
+    except asyncio.TimeoutError:
+        pass
+
+    goal_state = await juju_run(k8s_cp, "goal-state --format=json")
+    relation_status = json.loads(goal_state)
+    kube_api_endpoint = relation_status["relations"]["kube-api-endpoint"]
+    assert kube_api_endpoint["kubernetes-worker"]["status"] == "joined"
```

Comment:
this was "wait 5 minutes for the message `waiting_msg` to show up in the juju status on the k8s-control-plane application"

if that message didn't show up in 5 minutes -- it would raise a `TimeoutError`.  I had this happen to me twice while running the tests.  WHY?  Because the message came and went before the test framework had a chance to see it.  I guess it just missed the polling interval. 

So, I've added the except clause to catch the `TimeoutError`
and then check to see if the relation endpoint is actually `joined` after the 5 minutes or after the message appears once